common/ferrostar/src/navigation_controller/mod.rs

```diff
@@ -250,3 +253,48 @@ impl NavigationController {
         }
     }
 }
+
+/// JavaScript wrapper for `NavigationController`.
+#[cfg(feature = "wasm-bindgen")]
+#[wasm_bindgen(js_name = NavigationController)]
+pub struct JsNavigationController(NavigationController);
+
+#[cfg(feature = "wasm-bindgen")]
+#[wasm_bindgen(js_class = NavigationController)]
+impl JsNavigationController {
+    #[wasm_bindgen(constructor)]
+    pub fn new(route: JsValue, config: JsValue) -> Result<JsNavigationController, JsValue> {
+        let route: Route = serde_wasm_bindgen::from_value(route)?;
+        let config: NavigationControllerConfig = serde_wasm_bindgen::from_value(config)?;
+
+        Ok(JsNavigationController(NavigationController::new(
+            route, config,
+        )))
+    }
+
+    pub fn get_initial_state(&self, location: JsValue) -> Result<JsValue, JsValue> {
```

Comment:
I'm curious if there is a better way to avoid this newtype dance and return type erasure with `JsValue` on both sides. Let me know if you have any thoughts + maybe we can discuss on the call later.

Reply:
This was my biggest concern and a significant blocker during my early work. Initially, I tried to reuse the existing `uniffi::export` blocks, but `Arc`s don't make sense in JavaScript. Implementing `WasmDescribe` trait seems like a possible approach, but the workload could be insane. And according to wasm-bindgen's documentation, using `JsValue` is the recommended way to export structs to JavaScript.

Reply:
Gotcha; thanks! I'll do a bit of thinking on whether there's a better long-term solution. This feels like a case where we should be able to use a macro to reduce the boilerplate, but it's good enough as-is to merge. @Archdoog and I will experiment with other approaches.